### PR TITLE
Replace FilterList with std::vector<Filter> for future maintenance

### PR DESCRIPTION
**Pull request by cherylEnkidu**
#no-changelog

### CODE REVIEW

File: Firestore/core/test/unit/core/query_test.cc [hunk 1]
```diff
@@ -48,7 +49,6 @@ using testutil::CollectionGroupQuery;
 using testutil::DbId;
 using testutil::Doc;
 using testutil::Field;
-using testutil::Filter;
```
Comment (ehsannas) [inline Firestore/core/test/unit/core/query_test.cc:51]: why did you remove this?

**Review by ehsannas — COMMENTED** [review]
Generally LGTM. Not clear why the changes were made to query_test.cc (adding `testutil::` everywhere)?

File: Firestore/core/test/unit/core/query_test.cc (hunk 1 above)
Comment (cherylEnkidu) [reply Firestore/core/test/unit/core/query_test.cc:51]: Since we introduce `std::vector<Filter>` to this file. Here Filter refers to `core::Filter` (class name) instead of `testutil::Filter` (function name). With `using testutil::Filter;`, the compiler will not work.

**Review by ehsannas — APPROVED** [review]
